modules/nf-core/goleft/indexcov/main.nf

```diff
@@ -0,0 +1,54 @@
+
+process GOLEFT_INDEXCOV {
+    tag 'N=${bams.size()}'
+    label 'process_single'
+
+    conda "bioconda::goleft=0.2.4"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/goleft:0.2.4--h9ee0642_1':
+        'biocontainers/goleft:0.2.4--h9ee0642_1' }"
+
+    input:
+    tuple val(meta),path(bams),path(indexes)
+    path  fai
+
+    output:
+    tuple val(meta), path("${prefix}/*")  , emit: output
+    path "versions.yml"                   , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args0 = meta.args ?: ''
+    def args1 = task.ext.args ?: ''
```

Comment:
why do you need two args?

Reply:
@FriederikeHanssen  well, I find it strand to use task.ext to specify something that can be specific to a fasta sequence. For example in my private implementation of the module, I use the following line to set the sexual  and the excluded chromosomes. 

```
		--excludepatt `awk -F '\t' '(!(\$1 ~ /^(chr)?[0-9XY]+\$/)) {print \$1;}' "${fasta}.fai" | paste -s -d '|' `  \
		--sex `awk -F '\t' '(\$1 ~ /^(chr)?[XY]\$/) {print \$1;}' "${fasta}.fai" | paste -s -d, ` \
```

but don't use much task.ext so I'm not sure if it makes sense (?)

Reply:
I see. In general several args are reserved for when several tools are used (like bwa + samtools). In this case though I am wondering if it would make sense to set this commands in a different way. They look something that could get very easily go wrong when someone uses the module 🙈 @nf-core/maintainers do we have an example for something like this somewhere?

Reply:
This could be configured in the modules.config level, along the line of `ext.args = { meta.args }` (or something based on the meta map in any way), that is our standard for the modules.

Reply:
true, but I think in this particular case some extra docs could be good :D . Maybe not, let's see.